pkgs/development/interpreters/gnudatalanguage/default.nix

```diff
@@ -0,0 +1,159 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+# Native build inputs for the compiler
+, cmake
+, enableGAppsWrapping ? true
+, wrapGAppsHook ? null
+# Build dependencies
+, readline
+, ncurses #? curses
+, zlib
+, gsl
+, openmp
+, graphicsmagick #? imagemagick
+, netcdf
+, hdf4
+, hdf5
+, fftw, fftwFloat, fftwLongDouble
+, proj
+, shapelib
+, expat
+, udunits
+, eigen
+, pslib # a C library for PostScript
+, eccodes
+, glpk
+, libpng
+, plplot
+, libtiff
+, libgeotiff
+, libjpeg # for hdf4 and hdf5
+, libtirpc
+, pythonInterpreter ? python3, python3 ? null
+, enableMPI ? (stdenv.isLinux || stdenv.isDarwin)
+, libraryMPI ? mpich #? openmpi
+, mpich ? null
+# Unfree optional dependency
+# for hdf4 and hdf5
+, enableSzip ? false
+, szip ? null
+, enableWX ? true
+, wxWidgets ? wxGTK31, wxGTK31 ? null
+# X11: OFF by default since wxWidgets SHOULD take preference in ALL builds
+# https://github.com/gnudatalanguage/gdl/blob/master/CMakeLists.txt#L437
+, enableXWin ? false, xorg ? null
+}:
+
+assert enableWX -> wxWidgets != null;
+assert enableXWin -> (xorg != null && xorg.libX11 != null);
+assert enableGAppsWrapping -> (wrapGAppsHook != null);
+assert enableMPI -> (libraryMPI != null);
+assert enableSzip -> (szip != null);
+
+let
+  enablePlplotDrivers = enableWX || enableXWin;
+  plplot-with-drivers = plplot.overrideAttrs (oldAttrs: {
+    buildInputs =
+    (oldAttrs.buildInputs or [])
+    ++ (lib.optional enableWX wxWidgets)
+    ++ (lib.optional enableXWin xorg.libX11);
+  });
+in stdenv.mkDerivation rec {
+  pname = "gnudatalanguage";
+  version = "1.0.0";
+
+  src = fetchFromGitHub {
+    owner = pname;
+    repo = "gdl";
+    rev = "v${version}";
+    sha256 = "sha256-Y9LVRaWjQqpWqjNngxB406PE/rl/9S8rs0u0CK5ivUA=";
+  };
+
+  buildInputs = [
+    readline
+    ncurses
+    zlib
+    gsl
+    openmp
+    graphicsmagick
+    netcdf
+    (if enableSzip
+      then (hdf4.override { inherit szip; })
+      else hdf4)
+    (if enableSzip
+      then (hdf5.override { inherit szip; })
+      else hdf5)
+    fftw fftwFloat fftwLongDouble
+    proj
+    shapelib
+    expat
+    libraryMPI
+    udunits
+    eigen
+    pslib
+    eccodes
+    glpk
+    libpng
+    (if enablePlplotDrivers
+      then plplot-with-drivers
+      else plplot)
```

Comment:
Please move this to the end instead of inlining it.

Reply:
Done.